### PR TITLE
unit test for date filter on submission trend

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
*Description of changes:*
- changes for adding unit test on date filter on submission trend
- refactoring code to create a new method to check date filter parameters

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: analytics-api/src/analytics_api/models/user_response_detail.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class UserResponseDetail(BaseModel):  # pylint: disable=too-few-public-methods
     __tablename__ = 'user_response_detail'
 
     id = db.Column(db.Integer, primary_key=True, nullable=False)
-    survey_id = db.Column(db.Integer, ForeignKey('survey.id', ondelete='CASCADE'), primary_key=True, nullable=False)
```
Comment (tiagograf-AOT) [inline analytics-api/src/analytics_api/models/user_response_detail.py:19]: I think you are missing the migration to remove the primary key?

**Review by tiagograf-AOT — APPROVED** [review]

File: analytics-api/src/analytics_api/models/user_response_detail.py (hunk 1 above)
Comment (VineetBala-AOT) [reply analytics-api/src/analytics_api/models/user_response_detail.py:19]: somehow its not there in the migration file and was only in the model.

File: analytics-api/src/analytics_api/models/user_response_detail.py [hunk 2]
```diff
@@ -80,10 +78,20 @@ def get_response_count_by_created_week(
...
         return jsonify(result)
+
+    @staticmethod
+    def _append_search_options_filters(filters, search_options):
+        if search_options.get('from_date'):
+            filters.append(cast(UserResponseDetail.created_date, Date) >= search_options.get('from_date'))
+
+        if search_options.get('to_date'):
+            filters.append(cast(UserResponseDetail.created_date, Date) <= search_options.get('to_date'))
+
```
Comment (saravanpa-aot) [inline analytics-api/src/analytics_api/models/user_response_detail.py:96]: i think we dont need cast..
also check if  if 'from_date' in search_options:

File: analytics-api/tests/utilities/factory_utils.py [hunk 3]
```diff
@@ -64,3 +67,36 @@ def factory_email_verification_model(emailinfo: dict = TestEmailVerificationInfo
     db.session.add(emailverification)
     db.session.commit()
     return emailverification
+
+
+def factory_user_response_detail_model(survey_id, 
+                                       userresponseinfo: dict = TestUserResponseDetailInfo.userresponsedetail1):
+    """Produce a user response detail verification model."""
+    userresponsedetail = UserResponseDetailModel(
```
Comment (saravanpa-aot) [inline analytics-api/tests/utilities/factory_utils.py:75]: variable name userresponsedetail 
have underscrores plz

**Review by saravanpa-aot — APPROVED** [review]
some comments

File: analytics-api/src/analytics_api/models/user_response_detail.py (hunk 2 above)
Comment (VineetBala-AOT) [reply analytics-api/src/analytics_api/models/user_response_detail.py]: updated the if condition

File: analytics-api/tests/utilities/factory_utils.py (hunk 3 above)
Comment (VineetBala-AOT) [reply analytics-api/tests/utilities/factory_utils.py]: updated

**Pull request merged**